src/wayland/text_input_manager/mod.rs

```diff
@@ -0,0 +1,194 @@
+//! Utilities for text input support
+//!
+//! This module provides you with utilities to handle text input surfaces,
+//! it must be used in conjunction with the input method module to work.
+//!
+//! ```
+//! # extern crate wayland_server;
+//! use smithay::{
+//!     delegate_seat, delegate_tablet_manager, delegate_input_method_manager,
+//!     delegate_text_input_manager,
+//! };
+//! use smithay::wayland::seat::{Seat, SeatState, SeatHandler, XkbConfig};
+//! use smithay::wayland::input_method_manager::{
+//!     InputMethodManagerState,
+//!     InputMethodSeatTrait
+//! };
+//! use smithay::wayland::text_input_manager::{
+//!     TextInputManagerState,
+//!     TextInputSeatTrait
+//! };
+//!
+//! # struct State { seat_state: SeatState<Self> };
+//!
+//! delegate_seat!(State);
+//! delegate_input_method_manager!(State);
+//! // Delegate text input handling for State to TextInputManagerState.
+//! delegate_text_input_manager!(State);
+//!
+//! # let mut display = wayland_server::Display::<State>::new().unwrap();
+//! # let display_handle = display.handle();
+//!
+//! let seat_state = SeatState::<State>::new();
+//!
+//! // implement the required traits
+//! impl SeatHandler for State {
+//!     fn seat_state(&mut self) -> &mut SeatState<Self> {
+//!         &mut self.seat_state
+//!     }
+//! }
+//!
+//! // Add the seat state to your state and create manager globals
+//! InputMethodManagerState::new::<State>(&display_handle);
+//! TextInputManagerState::new::<State>(&display_handle);
+//!
+//! // create the seat
+//! let seat = Seat::<State>::new(
+//!     &display_handle,          // the display
+//!     "seat-0",                 // the name of the seat, will be advertized to clients
+//!     None                      // insert a logger here
+//! );
+//!
+//! seat.text_input();
+//! // Add text input capabilities to a seat
+//! seat.add_input_method(XkbConfig::default(), 200, 25);
+//! // Add input method capabilities to a seat, , needed for text input to work
+//!
+//! ```
+//!
+
+use wayland_protocols::wp::text_input::zv3::server::{
+    zwp_text_input_manager_v3::{self, ZwpTextInputManagerV3},
+    zwp_text_input_v3::ZwpTextInputV3,
+};
+use wayland_server::{backend::GlobalId, Client, DataInit, Dispatch, DisplayHandle, GlobalDispatch, New};
+
+use crate::wayland::seat::Seat;
+
+pub use text_input::TextInputHandle;
+pub use text_input::TextInputUserData;
+
+use super::input_method_manager::InputMethodHandle;
+
+const MANAGER_VERSION: u32 = 1;
+
+mod text_input;
+
+/// Extends [Seat] with text input functionality
+pub trait TextInputSeatTrait {
```

Comment:
Same here, `TextInputSeat` is enough imo.

Reply:
fixed